data_store/db/entities.py

```diff
@@ -49,18 +49,26 @@ class Funding(BaseModel):
     project: Mapped["Project"] = sqla.orm.relationship(back_populates="funding_records")
 
     __table_args__ = (
+        # migration auto-generate does not pick up these constrainsts
+        # TODO: Create migration to apply these constraints
+        # I suspect ck_risk_register_programme_junction_id_or_project_id is not applied
         sqla.CheckConstraint(
-            or_(
-                and_(programme_junction_id.isnot(None), project_id.is_(None)),
-                and_(programme_junction_id.is_(None), project_id.isnot(None)),
+            expression.cast(
+                or_(
+                    and_(programme_junction_id.isnot(None), project_id.is_(None)),
+                    and_(programme_junction_id.is_(None), project_id.isnot(None)),
+                ),
+                sqla.Boolean,
             ),
             name="ck_risk_register_programme_junction_id_or_project_id",
         ),
+        # I suspect ck_funding_start_or_end_date is not applied
         # check that both start and end dates are not null at the same time
         sqla.CheckConstraint(
-            or_(start_date.isnot(None), end_date.isnot(None)),
+            "start_date IS NOT NULL OR end_date IS NOT NULL",
```

Comment:
question: why do we need to change to text here?

Reply:
`CheckConstraint` is not accepting of `or_`'s return type. That's why I have to do the `expression.cast` above, which I think it's an ugly, hard to understand, work around to satisfy the typing. So, here, rewriting it to a string seemed both easier to read and fixed the typing, as `str` is accepted.

When I saw `CheckConstraint` not approve `or_`'s return type, it made me wonder if the code even works. That's when I noticed that the database does not list these constraints at all. So I might as well have removed them.